### PR TITLE
Refactor context options setting and getting

### PR DESCRIPTION
**Pull request by jukkar**
Unify storing and loading of context options. Previously there were multiple functions doing the same thing. Fixed issue when getting txtime option (bool vs int variable).

Currently the code expects this:
* for bool value in options, user is expected to supply int variable and setting len sizeof(int)
* for uint8_t value, user is expected to supply uint8_t variable and setting len sizeof(uint8_t)
* for uint16_t value, user is expected to supply int variable and setting len sizeof(int)

### CODE REVIEW

File: subsys/net/ip/net_context.c [hunk 1]
```diff
@@ -2817,6 +2821,21 @@ int net_context_set_option(struct net_context *context,
...
+		uint16_t opt16;
+		uint8_t opt8;
+		bool optbool;
+		int optint;
+
+		(void)set_uint16_option(&opt16, &optint, sizeof(int));
+		(void)set_uint8_option(&opt8, &opt8, sizeof(uint8_t));
+		(void)set_bool_option(&optbool, &optint, sizeof(int));
+	}
+
```
Comment (pdgendt) [inline subsys/net/ip/net_context.c:2838]: Why not add `__maybe_unused` to those functions?

**Comment by pdgendt** [comment]
Nit: duplicate commit subject for getter/setter changes

File: subsys/net/ip/net_context.c (hunk 1 above)
Comment (jukkar) [reply subsys/net/ip/net_context.c:2838]: Hmm, why not indeed. I did not knew about such attribute, but will change the code to use that.

**Comment by jukkar** [comment]
Are we using the length parameter correctly in the option functions? What I mean is that Instead of rejecting the get/set if the length is not exactly what we expect, we could (at least in theory) figure out how we could cast the value parameter so that value is correctly accessed. WDYT?

**Comment by jukkar** [comment]
Updated according to comments.

**Review by pdgendt — APPROVED** [review]

**Comment by rlubos** [comment]
> Are we using the length parameter correctly in the option functions? What I mean is that Instead of rejecting the get/set if the length is not exactly what we expect, we could (at least in theory) figure out how we could cast the value parameter so that value is correctly accessed. WDYT?

IMO, whether we store the option value as bool or some smaller integer is just an internal implementation detail. If we follow the socket convention, the man page for setsockopt() says that in most cases an `int` is expected:
> Most socket-level options utilize an int argument for optval.

I'm not really convinced there is a need to change that.

**Review by rlubos — APPROVED** [review]

**Comment by jukkar** [comment]
> IMO, whether we store the option value as bool or some smaller integer is just an internal implementation detail. If we follow the socket convention, the man page for setsockopt() says that in most cases an `int` is expected:
> 
> > Most socket-level options utilize an int argument for optval.
> 
> I'm not really convinced there is a need to change that.

I am not proposing anything major changes, just trying to make the getters/setters a bit more robust. I will experiment a bit and propose a separate PR for review.

**Pull request merged**